packages/es-components/src/components/patterns/datepicker/DatePicker.js

```diff
@@ -30,25 +30,55 @@ class DateTextbox extends React.Component {
   }
 }
 
-const NativeDatePicker = props => {
-  const onChangeIntercept = event => props.onChange(moment(event.target.value));
+const getVerifiedDate = selectedDate => {
+  let verifiedDate = selectedDate;
+  if (typeof selectedDate === 'string') {
+    const newDateMoment = moment(selectedDate, 'L');
+    const isFullDate =
+      selectedDate && newDateMoment.format('L') === selectedDate;
+    verifiedDate = isFullDate ? newDateMoment : undefined;
```

Comment:
With react 16.8 released I think we should rework this PR after fixing #260 since we remove moment in favor of `date-fns`.

Reply:
I don't think this should wait for that. If we do, it'll mean forcing us to migrate a couple of other repositories before we can have this bug fix in place. If it helps, I can make a PR that resolves any conflicts this one creates on the `next` branch right after we merge this.

Reply:
Fair enough.